transform/transformers/mbs_transformer.py

```diff
@@ -0,0 +1,196 @@
+import datetime
+import decimal
+import json
+import logging
+import os
+from collections import namedtuple
+from decimal import ROUND_HALF_UP, Decimal
+
+from structlog import wrap_logger
+
+from transform.settings import (
+    SDX_FTP_DATA_PATH,
+    SDX_FTP_IMAGE_PATH,
+    SDX_FTP_RECEIPT_PATH,
+    SDX_RESPONSE_JSON_PATH,
+)
+from transform.transformers.cs_formatter import CSFormatter
+from transform.transformers.survey import Survey
+from transform.transformers.transformer import ImageTransformer
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+# Set the rounding contect for Decimal objects to ROUND_HALF_UP
+decimal.getcontext().rounding = ROUND_HALF_UP
+
+
+class MBSTransformer():
+    """Perform the transforms and formatting for the MBS survey."""
+
+    Identifiers = namedtuple(
+        "Identifiers",
+        [
+            "batch_nr",
+            "seq_nr",
+            "ts",
+            "tx_id",
+            "survey_id",
+            "inst_id",
+            "user_ts",
+            "user_id",
+            "ru_ref",
+            "ru_check",
+            "period",
+        ],
+    )
+
+    @staticmethod
+    def _merge_dicts(x, y):
+        """Makes it possible to merge two dicts on Python 3.4."""
+        z = x.copy()
+        z.update(y)
+        return z
+
+    @staticmethod
+    def round_mbs(value):
+        """MBS rounding is done on a ROUND_HALF_UP basis and values are divided by 1000 for the pck"""
+        return Decimal(round(Decimal(float(value))) / 1000).quantize(1)
+
+    def __init__(self, response, seq_nr=0):
+
+        self.idbr_ref = {"0255": "MB65B"}
```

Comment:
Does this need to be a dict? Will the other MBS surveys have different entries?

Reply:
Yes - from what I can see different form types will map differently.